sdk/cosmosdb/cosmos/README.md

```diff
@@ -183,6 +183,18 @@ for (const city of resources) {
 
 For more information on querying Cosmos DB databases using the SQL API, see [Query Azure Cosmos DB data with SQL queries][cosmos_sql_queries].
 
+## Error Handling
+
+The SDK generates various types of errors that can occur during operation.
+
+1. `AbortError` is thrown if any user passed signal caused the abort.
+2. `Timeout Error` is thrown if Abort is called internally due to timeout.
+3. `ErrorResponse` is thrown if the response of an operation returns an error code of >=400.
+4. `RestError` is thrown in case of failure of underlying system call due to network issues. For Eg. for error codes `ETIMEDOUT`, `ESOCKETTIMEDOUT`, `ECONNREFUSED`, `ECONNRESET`, `ENOENT`, and `ENOTFOUND` this error is thrown.
+5. Errors generated by any devDependencies. For Eg. `@azure/identity` package could throw `CredentialUnavailableError`.
+
+It's important to properly handle these errors to ensure that your application can gracefully recover from any failures and continue functioning as expected.
```

Comment:
Something we've been trying to do for custom error types is provide typeguards that consumers can use inside catch clauses to strongly type the error shape. For example, we provide this one for `RestError`:

https://github.com/Azure/azure-sdk-for-js/blob/d83b1ddf0c363df02768bbb9922377d39ab3c95d/sdk/core/core-rest-pipeline/src/restError.ts#L93

Perhaps it would be useful to talk about these guards and show an example of using them?

Reply:
@xirzec @amanrao23 Do I need to install the core library to make use of the core error objects? Or are they exposed through the Cosmos DB SDK so only 1 npm package to install? 

In the past with typescript, I usually wind up installing that separate library to get the types.

Reply:
if they are re-exported by the service SDK you can avoid depending on core directly (which we prefer, honestly.)

I think we (the SDK team) should figure out some good guidelines around this, namely about re-exporting error types and type guards in client SDKs that use them. This dovetails with how we ask service client packages to re-export credential types from core-auth.